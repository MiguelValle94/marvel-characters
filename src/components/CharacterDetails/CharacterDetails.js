import React from 'react';
import PropTypes from 'prop-types';

export default function CharacterDetails({ character }) {
  return (
    <>
      {character && (
        <div className='card'>
          <img
            src={`${character.thumbnail.path}.${character.thumbnail.extension}`}
            className='card-img-top'
            alt={character.name}
          />
          <div className='card-body'>
            <h5 className='card-title'>{character.name}</h5>
          </div>
        </div>
      )}
    </>
  );
}

CharacterDetails.propTypes = {
  character: PropTypes.object.isRequired
};
