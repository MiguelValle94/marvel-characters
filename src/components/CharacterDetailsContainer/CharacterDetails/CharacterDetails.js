import React from 'react';
import PropTypes from 'prop-types';

export default function CharacterDetails({ character }) {
  return (
    <div className='card mb-3'>
      <div className='row no-gutters'>
        <div className='col-md-4'>
          <img
            src={`${character.thumbnail.path}.${character.thumbnail.extension}`}
            className='card-img'
            alt={character.name}
          />
        </div>
        <div className='col-md-8'>
          <div className='card-body'>
            <h5 className='card-title'>{character.name}</h5>
            <p className='card-text'>
              {character.description || 'No description available.'}
            </p>
            <p className='card-text'>
              <small className='text-muted'>
                Last updated {new Date(character.modified).toLocaleString()}
              </small>
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}

CharacterDetails.propTypes = {
  character: PropTypes.object.isRequired
};
