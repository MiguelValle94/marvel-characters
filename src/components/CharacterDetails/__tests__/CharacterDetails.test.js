import React from 'react';
import { shallow } from 'enzyme';

import CharacterDetails from '../CharacterDetails';
import { charactersMock } from '../../../mocks/characters.mock';

const setUpShallow = (character) => {
  const component = shallow(<CharacterDetails character={character} />);
  return component;
};

describe('CharacterDetails', () => {
  let component;

  it('should display characters list', () => {
    component = setUpShallow(charactersMock.data.results[0]);
    expect(component).toMatchSnapshot();
  });
});
