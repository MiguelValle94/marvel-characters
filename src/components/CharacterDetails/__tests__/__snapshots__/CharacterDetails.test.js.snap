// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetails should display characters list 1`] = `
<Fragment>
  <div
    className="card"
  >
    <img
      alt="3-D Man"
      className="card-img-top"
      src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784.jpg"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        3-D Man
      </h5>
    </div>
  </div>
</Fragment>
`;
